misc/window_control.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+from talon.voice import Context, Key
+
+ctx = Context("window_control")
+
+def jump_tab(m):
+    tab_number = parse_words_as_integer(m._words[1:])
+    if tab_number != None and tab_number > 0 and tab_number < 9:
+        press("cmd-%s" % tab_number)
+
+ctx.keymap(
+    {
+        # tab control
+        "new tab": Key("cmd-t"),
+        "close tab": Key("cmd-w"),
+        "(next tab | goneck)": Key("cmd-shift-]"),
+        "next tab": Key("cmd-shift-]"),
+        "((last | previous | preev) tab | gopreev)": Key("cmd-shift-["),
+        "tab (1 | 2 | 3 | 4 | 5 | 6 | 7 | 8)": jump_tab,
+        "(end | rightmost) tab": Key("cmd-9"),
+        # zooming
+        "zoom in": Key("cmd-="),
+        "zoom out": Key("cmd--"),
+        "zoom normal": Key("cmd-0"),
+        # window control
+        "new window": Key("cmd-n"),
+        "close window": Key("cmd-shift-w"),
+        "(next window | gibby)": Key("cmd-`"),
+        "(last window | shibby)": Key("cmd-shift-`"),
+        "(window space right | next space)": Key("ctrl-right"),
+        "(window space left | last space)": Key("ctrl-left"),
+        "curtail": Key("cmd-m"),
```

Comment:
What is this?

Reply:
minimise

Reply:
Maybe `(window minimize | curtail)` then?

Reply:
👍 also, should it be `window new`, `window next` etc. to keep the pattern?

Reply:
tab new, tab close?

Reply:
Ah yeah. Pretty sure aegis has suggested that standard: noun action

Reply:
hmm...but the generic editor script is doing action noun. and eg `word go left` , `line delete left` sounds odd to me. maybe the principle is that whichever word is most uncommon, whether action or noun, should be first

Reply:
`word go left` is actually ok `noun verb adverb`. Line delete left sounds alright, little awkward. You might check slack, pretty sure aegis has a pretty strong opinion.

Reply:
didn't find aegis' comment, but https://talonvoice.slack.com/archives/CARRXTUN4/p1549664411062700 suggests that `verb noun` is the way to go